api/src/integrations/epic/base-record-utils.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const defaultLog = require('../../utils/logger')('epic-base-record-utils');
+const EpicUtils = require('./epic-utils');
+const DocumentController = require('./../../controllers/document-controller');
+
+/**
+ * Epic base record type handler that can be used directly, or extended if customizations are needed.
+ *
+ * Must contain the following functions:
+ * - transformRecord: (object) => object
+ * - saveRecord: (object) => any
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+  }
+
+  /**
+   * Persist an epic record to the database, and return an array of ObjectIds.
+   *
+   * @param {*} epicRecord Epic record
+   * @returns {Array<string>} array of ObjectIds
+   * @memberof BaseRecordUtils
+   */
+  async saveDocument(epicRecord) {
+    const documents = [];
+
+    if (epicRecord && epicRecord._id && epicRecord.documentFileName) {
+      const savedDocument = await DocumentController.createDocument(
+        epicRecord.documentFileName,
+        (this.auth_payload && this.auth_payload.displayName) || '',
+        `https://projects.eao.gov.bc.ca/api/document/${epicRecord._id}/fetch/${encodeURIComponent(
```

Comment:
Hostname of this part can be moved up to the top of the page as a const

Reply:
I will fix this now

Reply:
Updated